csrc/ir/nodes.cpp

```diff
@@ -3079,6 +3079,35 @@ TensorDomain::TensorDomain(
   resetDomains();
 }
 
+TensorDomain::TensorDomain(
+    IrBuilderPasskey passkey,
+    std::vector<IterDomain*> root_domain,
+    std::vector<int64_t> stride_order,
+    std::vector<std::optional<bool>> contiguity)
+    : Val(passkey, ValType::TensorDomain, DataType::Null),
+      root_domain_(std::move(root_domain)),
+      leaf_domain_(root_domain_),
+      contiguity_(
+          contiguity.empty() ? getContiguityFilledWith(maybeAllocation(), false)
+                             : std::move(contiguity)),
+      has_reduction_(false) {
+  // setting the proper allocation domain
+  if (!stride_order.empty()) {
+    auto rank = root_domain_.size();
+    NVF_ERROR(
+        rank == stride_order.size(), "Invalid size of stride_order vector");
```

Comment:
Does `stride_order.size()` have to equal `root_domain_.size()`? I thought it's totally fine for, e.g., a ViewOp to convert a root domain of size 1 to an rfactor domain and an allocation domain of size 2.

Reply:
Not necessary in general. But for this API, we are given only root_domain and stride_order. So allocation domain is supposed to be a permutation on root_domain.

The use case you were mentioning, we probably want to use the other TensorDomain constructor where you can explicit provide root_domain/rfactor_domain/leaf_domain/allocation_domain